Makefile

```diff
@@ -107,9 +107,19 @@ ifndef $(GOPATH)
 endif
 
 # -- file lists
-ARGOEXEC_PKG_FILES        := $(shell go list -f '{{ join .Deps "\n" }}' ./cmd/argoexec/ |  grep 'argoproj/argo-workflows/v3/' | xargs go list -f '{{ range $$file := .GoFiles }}{{ print $$.ImportPath "/" $$file "\n" }}{{ end }}' | cut -c 39-)
-CLI_PKG_FILES             := $(shell go list -f '{{ join .Deps "\n" }}' ./cmd/argo/ |  grep 'argoproj/argo-workflows/v3/' | xargs go list -f '{{ range $$file := .GoFiles }}{{ print $$.ImportPath "/" $$file "\n" }}{{ end }}' | cut -c 39-)
-CONTROLLER_PKG_FILES      := $(shell go list -f '{{ join .Deps "\n" }}' ./cmd/workflow-controller/ |  grep 'argoproj/argo-workflows/v3/' | xargs go list -f '{{ range $$file := .GoFiles }}{{ print $$.ImportPath "/" $$file "\n" }}{{ end }}' | cut -c 39-)
+HACK_PKG_FILES_AS_PKGS ?= false
+ifeq ($(HACK_PKG_FILES_AS_PKGS),false)
```

Comment:
Would it make more sense to detect Windows instead of having the Windows Dockerfile use an env var?

Reply:
We should also add a comment here, since out of context this would be very confusing to a reader

Reply:
If we can detect Windows and do this the long way around we could attempt a mklink there too.

If I get access to some kind of windows that let's me play I might have a go, but I'd rather right now it was awkward and people who can fix it found this and tried to fix it.

Reply:
You can definitely detect Windows, in the same way you can detect a Linux OS, e.g. [with `uname` or `$OSTYPE`](https://stackoverflow.com/q/394230/3431180). There are some variations on that (i.e. prior to WSL etc when Cygwin and MinGW were more used), but technically the main one we need to handle is the GH + Docker one, the rest a Windows contributor could figure out